protocol/synthetix/contracts/modules/core/CollateralModule.sol

```diff
@@ -72,7 +72,12 @@ contract CollateralModule is ICollateralModule {
             .load(collateralType)
             .convertTokenToSystemAmount(tokenAmount);
 
-        if (account.collaterals[collateralType].availableAmountD18 < systemAmount) {
+        (uint256 totalDeposited, uint256 totalAssigned, uint256 totalLocked) = account
+            .getCollateralTotals(collateralType);
+        if (
+            totalDeposited - totalAssigned < systemAmount ||
```

Comment:
this condition should probably just be `if (totalDeposited - totalAssigned - totalLocked)` because `totalDeposited` is literally `totalAssignedD18 + self.collaterals[collateralType].availableAmountD18`

Reply:
I see. Will update.

Reply:
Actually, loading `availableAmount` from storage a second time is more expensive than reading memory variables and extra math op

Reply:
I just realized that locked collateral can also be assigned. So really, it should be the greater of the two `totalLocked` and `totalAssigned` is what we subtract by

Reply:
Is all locked collateral not assigned?
ref:
```
@return totalAssigned The amount of collateral in the account that is delegated to pools, denominated with 18 decimals of precision.
@return totalLocked The amount of collateral in the account that cannot currently be undelegated from a pool, denominated with 18 decimals of precision.
```

Reply:
if so:
```
if (totalDeposited - totalLocked < systemAmount) {
            revert InsufficientAccountCollateral(systemAmount);
        }
```